frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx

```diff
@@ -1,37 +1,81 @@
 import * as React from 'react';
 import * as _ from 'lodash';
+import { safeLoadAll } from 'js-yaml';
 import { match as RMatch } from 'react-router';
 import {
   navFactory,
   LoadingBox,
   StatusBox,
   FirehoseResult,
 } from '@console/internal/components/utils';
+import { coFetchJSON } from '@console/internal/co-fetch';
 import { SecretModel } from '@console/internal/models';
 import { ErrorPage404 } from '@console/internal/components/error';
 import { DetailsPage } from '@console/internal/components/factory';
-import { K8sResourceKindReference } from '@console/internal/module/k8s';
+import { K8sResourceKindReference, K8sResourceKind } from '@console/internal/module/k8s';
 import HelmReleaseResources from './HelmReleaseResources';
 import HelmReleaseOverview from './HelmReleaseOverview';
+import { HelmRelease } from './helm-types';
 
 const SecretReference: K8sResourceKindReference = 'Secret';
 const HelmReleaseReference = 'HelmRelease';
 export interface HelmReleaseDetailsPageProps {
   match: RMatch<{
     ns?: string;
+    name?: string;
   }>;
   secret?: FirehoseResult;
 }
 
 const HelmReleaseDetailsPage: React.FC<HelmReleaseDetailsPageProps> = ({ secret, match }) => {
   const namespace = match.params.ns;
+  const helmReleaseName = match.params.name;
+  const helmManifestResources = React.useRef([]);
```

Comment:
Why isn't this `useState`?

Reply:
I didn't wanted `helmManifestResources` as a depepndency in the `useEffect` dependency array. That was leading to unnecesarry renders and side effects.

Reply:
My confusion is in what's triggering the re-render when the resources are updated?

Reply:
I've actually updated the PR to use `useState`  instead of `useRef` . For some reason earlier `MultiListPage` was misbehaving when I tested it with `useState` . You could see no resources message for a second and then it would update to show the loaded resources. Its working fine now.